lib/livebook/zta/teleport.ex

```diff
@@ -0,0 +1,90 @@
+defmodule Livebook.ZTA.Teleport do
+  use GenServer
+  require Logger
+
+  defstruct [:name, :req_options, :jwks]
+  @fields %{"sub" => :id, "username" => :username}
+
+  @assertion "teleport-jwt-assertion"
+  def start_link(opts) do
+    url = opts[:identity][:key]
+    options = [req_options: [url: url]]
+    GenServer.start_link(__MODULE__, options, name: opts[:name])
+  end
+
+  @impl true
+  def init(options) do
+    state = struct!(__MODULE__, options)
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_call(:get_jwks, _, state) do
+    try do
+      jwks =
+        Req.request!(state.req_options).body["keys"]
```

Comment:
Doing this inside the GenServer will be a bottleneck. Instead, invoke the GenServer only to get the cluster address, and call Req from the authenticate callback.

Reply:
My idea here was to cache the `keys`. 

I should actually change the code to
```elixir
def handle_call(:get_jwks, _, %{jwks: nil} = state) do
...
end
def handle_call(:get_jwks, _, state) do
{:noreply, state.jwks, state}
end
```

What do you think?

Reply:
If we cache it, we need to probably set up the mechanism to expire it.

Reply:
@josevalim the other providers we query in the server and `Process.send_after(self(), :renew, @renew_afer)`, so we can do the same for consistency?

Reply:
Then it sounds good to me, but without the try catch. :)